lib/parse-with-scope.js

```diff
@@ -0,0 +1,19 @@
+"use strict";
+
+const t = require("@babel/types");
+const analyzeScope = require("./analyze-scope");
+const parse = require("./parse");
+
+module.exports = function(code, options) {
+  options = options || {};
+  options.ecmaVersion = options.ecmaVersion || 6;
```

Comment:
I don’t think this is used anywhere else (it looks unused in the parse method that is called with these defined options). Babylon doesn’t really have a concept of ES versions - it always is able to parse everything.

Reply:
This is to rewrite the options object directly because some core rules depends on `parserOptions.ecmaVersion`.

Reply:
Ah I see. In that case, We should probably set this to the latest version ESLint core rules support.

Reply:
As discussed in the ESLint Gitter, I'll make a followup PR for this.